python/ray/train/lightgbm/lightgbm_trainer.py

```diff
@@ -126,9 +126,7 @@ def _model_iteration(
             return model.current_iteration()
         return model.booster_.current_iteration()
 
-    def preprocess_datasets(self) -> None:
-        super().preprocess_datasets()
-
+    def setup(self) -> None:
```

Comment:
nit: Should this be in `GBDTTrainer` instead? Seems like both LightGBM and XGBoost have the same definition?